kiosk/src/Components/AddingGame.tsx

```diff
@@ -58,60 +61,82 @@ const AddingGame: React.FC<IProps> = ({ kiosk }) => {
         };
     });
 
+    const pollForGames =  (kioskCode: string) => {
+        const timeoutDuration = 570000; // wait for 9.5 minutes until the kiosk code expires
+        const whenToPoll = 5000; // wait for 5 seconds to poll for game code data
+        if (kiosk.state !== KioskState.AddingGame) {
+            return;
+        }
+        return new Promise<void>(async (resolve, reject) => {
+            const getGameCode = async () => {
+                try {
+                    const gameCode: string = await getGameCodeAsync(kioskCode);
+                    await kiosk.saveNewGameAsync(gameCode);
+                    clearTimeout(pollFrequency);
+                    clearTimeout(pollTimeout);
+                    resolve();
+                    kiosk.launchGame(gameCode);
+                }
+                catch (error: any) {
+                    if (error.message === "Not Found" || 
```

Comment:
Would it be possible to make the timer reset dependent on knowing what the returned error was rather than knowing it wasn't in a certain set? e.g.
```ts
if (error.message === "The error I know means it's ok to poll again") {
    // setup poll timer for another run
} else {
    // cleanup and return
}
```